gen/lib/gen/generator.rb

```diff
@@ -0,0 +1,115 @@
+require 'gen'
+require 'fileutils'
+require 'pathname'
+
+module Gen
+  class Generator
+    def self.run(project_name)
+      new(project_name).run
+    end
+
+    TEMPLATE_ROOT = File.expand_path('gen/template', Gen::ROOT)
+
+    VALID_PROJECT_NAME = /\A[a-z][a-z0-9]*\z/
+    private_constant :VALID_PROJECT_NAME
+
+    # false  -> delete file
+    # string -> rename file before applying template substitutions
+    VENDOR_TRANSLATIONS = {
+      'Gemfile'            => false,
+      'exe/__app__-gems'   => false,
+      'exe/__app__-vendor' => 'exe/__app__',
```

Comment:
`dev.yml` has bundler and should be excluded here, or else alternated like `'exe/__app__-*'`